.travis.yml

```diff
@@ -31,7 +32,7 @@ cache:
     - dist-newstyle
     - bin
 
-install:
```

Comment:
Haskell projects in Travis have default commands for `install` and `script` (https://docs.travis-ci.com/user/languages/haskell/) which don't work here and need to be overridden. If it's not broken, don't fix it :-)

Reply:
Unfortunately, it seems that Travis runs `cabal update` before running the `install` phase (and also the `before_install` phase). With version 3.2 of `cabal-install`, `cabal update` is an alias for `cabal new-update`, which will fail if paths mentioned in `cabal.project` do not contain `.cabal` files.

I'm going to see if we can disable this implicit `cabal update` somehow, and I'll move the submodule clones back to `install` since `before_install` doesn't do the right thing.